internal/ringbuffer.go

```diff
@@ -0,0 +1,119 @@
+package internal
+
+import (
+	"fmt"
+	"github.com/hazelcast/hazelcast-go-client/core"
+	"github.com/hazelcast/hazelcast-go-client/internal/common/collection"
+	. "github.com/hazelcast/hazelcast-go-client/internal/protocol"
+)
+
+type RingbufferProxy struct {
+	*partitionSpecificProxy
```

Comment:
In java this is not `partitionSpecificProxy`, as far as I see in some clients this is `partititionSpecificProxy` and in some it is not, what should this be?

Reply:
I took Node.js as reference.

Reply:
This is from Javadoc:
```
 * A Ringbuffer currently is a replicated but not partitioned data-structure. So all data is stored in a single partition;
 * comparable to the IQueue implementation. But we'll provide an option to partition the data in the near future.
```
In practice, it is partition specific. If it becomes partitioned in the future, we can change this easily. It is implementation detail anyway.